jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/DataCloudStatement.java

```diff
@@ -48,21 +43,30 @@ public class DataCloudStatement implements Statement, AutoCloseable {
     protected static final String BATCH_EXECUTION_IS_NOT_SUPPORTED =
             "Batch execution is not supported in Data Cloud query";
     protected static final String CHANGE_FETCH_DIRECTION_IS_NOT_SUPPORTED = "Changing fetch direction is not supported";
-    private static final String QUERY_TIMEOUT = "queryTimeout";
 
-    private int queryTimeout;
+    private StatementProperties statementProperties;
+
+    /**
+     * The target maximum number of rows for a query. The default means disabled.
+     */
+    @Getter(AccessLevel.PACKAGE)
+    private int targetMaxRows = 0;
+
+    /**
+     * The target maximum number of bytes per RPC response. This is only relevant when `targetMaxRows > 0`.
+     * The default means disabled.
+     */
+    @Getter(AccessLevel.PACKAGE)
+    private int targetMaxBytes = 0;
 
     public DataCloudStatement(@NonNull DataCloudConnection connection) {
         this.connection = connection;
-        val properties = connection.getClientInfo();
-        this.queryTimeout = getIntegerOrDefault(properties, QUERY_TIMEOUT, DEFAULT_QUERY_TIMEOUT);
-        this.targetMaxBytes = getIntegerOrDefault(properties, BYTE_LIMIT, HYPER_MAX_ROW_LIMIT_BYTE_SIZE);
+        statementProperties = connection.getConnectionProperties().getStatementProperties();
```

Comment:
for consistency with line above:

```suggestion
        this.statementProperties = connection.getConnectionProperties().getStatementProperties();
```